### ISSUE TITLE
EVL103 (unused argument) shouldn't warn for args with "unused_" prefix

### ISSUE THREAD

**Issue by dbarnett** [syngan/vim-vimlint#101]
The "unused argument" warning shouldn't trigger for an argument that explicitly indicates it's unused with an "unused_" prefix. That's a common, unambiguous convention and more readable than `@vimlint` comments.

For example, I see ```[EVL103] unused argument `a:unused_arg` ``` for a function like:
```vim
function! SomeCompletionList(unused_arg, line, pos) abort
  " Do something with a:line and a:pos…
endfunction
```
but I'd like to see no warning for that function.

**Comment by blueyed** [comment]
Is `unused_` as a prefix more common than `_`?

You can ignore only `_` currently by using `-e 'EVL103.a:_=1'`, but it would be nice to ignore the pattern `_*` also, i.e. `_arg` should not trigger EVL103.

**Comment by blueyed** [comment]
@syngan 
Do you think support for `-e 'EVL103.a:_*=1' would make sense, i.e. allow for glob patterns in there ("_*" matching everything that starts with an underscore)?

**Comment by dbarnett** [comment]
I can't say what's more common "globally", but `unused_` prefix is the preferred signal in certain massive codebases where batch-changing everything to a different convention to suppress junk lint warnings wouldn't be an option.

**Comment by syngan** [comment]
@blueyed  `-e 'EVL103.a:_.*=1' is supported.

**Comment by blueyed** [comment]
@syngan 
Awesome!
Is this somethind added only recently, or have I not tried it correctly before?

@dbarnett 
Therefore `-e 'EVL103.a:unused_.*=1'` should work for you then, and this issue can be closed.

**Comment by dbarnett** [comment]
I was able to shut up these warnings coming from syntastic by adding these lines into my vimrc:
```vim
let g:syntastic_vimlint_options = {'EVL103': {'a:unused.*' : 1}}
let g:vimlint#config = g:syntastic_vimlint_options
```

That's something at least. However, as a project maintainer there doesn't seem to be any simple way to configure vimlint to not give a bunch of junk warnings on my project (e.g. vimlint default behavior or config I can add to my project). I'll check out the new vim linting tool https://github.com/Kuniwak/vint to see if it does better for me.

**Comment by blueyed** [comment]
vint is good to check out, and has enabled checking unused args by default in 0.4a1 now.

As for Neomake I am using a Makefile to give those arguments, e.g. vimlint is meant to be run using `make vimlint`.
For live checking I am only using vint (via Neomake), since it is also much faster.

btw: you might want to try Neomake instead of Syntastic, because it is async.

**Comment by syngan** [comment]
> Is this somethind added only recently, or have I not tried it correctly before?

I think it is the latter. I implemented it in 2013.
